mlfinlab/data_structures/base_bars.py

```diff
@@ -0,0 +1,187 @@
+"""
+A base class for the various bar types. Includes the logic shared between classes, to minimise the amount of
+duplicated code.
+"""
+
+from abc import ABC, abstractmethod
+
+import pandas as pd
+import numpy as np
+
+
+class BaseBars(ABC):
+    """
+    Abstract base class which contains the structure which is shared between the various standard and information
+    driven bars. There are some methods contained in here that would only be applicable to information bars but
+    they are included here so as to avoid a complicated nested class structure.
+    """
+
+    def __init__(self, file_path, metric, batch_size=2e7):
+        """
+        Constructor
+
+        :param file_path: (String) Path to the csv file containing raw tick data in the format[date_time, price, volume]
+        :param metric: (String) type of imbalance bar to create. Example: dollar_imbalance.
+        :param batch_size: (Int) Number of rows to read in from the csv, per batch.
+        """
+        # Base properties
+        self.file_path = file_path
+        self.metric = metric
+        self.batch_size = batch_size
+        self.prev_tick_rule = 0
+
+        # Batch_run properties
+        self.flag = False  # The first flag is false since the first batch doesn't use the cache
+        self.cache = []
+
+    def batch_run(self, verbose=True, to_csv=False):
+        """
+        Reads a csv file in batches and then constructs the financial data structure in the form of a DataFrame.
+        The csv file must have only 3 columns: date_time, price, & volume.
+
+        :return: (DataFrame) Financial data structure
+        """
+        # Read in the first row & assert format
+        first_row = pd.read_csv(self.file_path, nrows=1)
+        self._assert_csv(first_row)
+
+        if verbose:  # pragma: no cover
+            print('Reading data in batches:')
+
+        # Read csv in batches
+        count = 0
+        final_bars = []
+        cols = ['date_time', 'open', 'high', 'low', 'close', 'volume']
+        for batch in pd.read_csv(self.file_path, chunksize=self.batch_size):
+            if verbose:  # pragma: no cover
+                print('Batch number:', count)
+
+            list_bars = self._extract_bars(data=batch)
+
+            if to_csv is True:
+                pd.DataFrame(list_bars, columns=cols).to_csv(
```

Comment:
Does this create many different csv files instead of updating a single one? Ideally we would only want a single csv file. Also where does the user specify where these files are written to?

Reply:
If so, please write a unit test to correct this as well.

Reply:
Add unit tests and class variable with file path